### PR TITLE
chore: move filter params into the clearCookies

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/api/class-browsercontext.md [hunk 1]
```diff
@@ -456,7 +456,71 @@ Returns the browser instance of the context. If it was launched as a persistent
...
+await context.clearCookies({ domain: 'my-origin.com' });
+await context.clearCookies({ domain: /.*my-origin\.com/ });
+await context.clearCookies({ path: '/api/v1' });
+await context.clearCookies({ name: 'session-id', domain: 'my-origin.com' });
+```
+
+
+```java
+context.clearCookies();
+context.clearCookies(new ClearCookies().setName("session-id"));
```
Comment (dgozman) [inline docs/src/api/class-browsercontext.md:475]:
```suggestion
context.clearCookies(new BrowserContext.ClearCookiesOptions().setName("session-id"));
```

File: docs/src/api/class-browsercontext.md [hunk 2]
```diff
@@ -456,7 +456,71 @@ Returns the browser instance of the context. If it was launched as a persistent
...
+context.clear_cookies()
+context.clear_cookies(name="session-id")
+context.clear_cookies(domain="my-origin.com")
+context.clear_cookies(path="/api/v1")
+context.clear_cookies(name="session-id", domain="my-origin.com")
+```
+
+```csharp
+await context.ClearCookiesAsync();
+await context.ClearCookiesAsync(new [] { Name = "session-id" });
```
Comment (dgozman) [inline docs/src/api/class-browsercontext.md:501]:
```suggestion
await context.ClearCookiesAsync(new () { Name = "session-id" });
```

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-browsercontext.md [hunk 3]
```diff
@@ -456,7 +456,71 @@ Returns the browser instance of the context. If it was launched as a persistent
...
+await context.ClearCookiesAsync(new [] { Domain = "my-origin.com" });
+await context.ClearCookiesAsync(new [] { Path = "/api/v1" });
+await context.ClearCookiesAsync(new [] { Name = "session-id", Domain = "my-origin.com" });
```
Comment (mxschmitt) [inline docs/src/api/class-browsercontext.md:504]:
```suggestion
await context.ClearCookiesAsync(new() { Domain = "my-origin.com" });
await context.ClearCookiesAsync(new() { Path = "/api/v1" });
await context.ClearCookiesAsync(new() { Name = "session-id", Domain = "my-origin.com" });
```

File: packages/playwright-core/src/server/browserContext.ts [hunk 4]
```diff
@@ -276,20 +276,25 @@ export abstract class BrowserContext extends SdkObject {
...
-
+  async clearCookies(options: {name?: string | RegExp, domain?: string | RegExp, path?: string | RegExp}): Promise<void> {
     const currentCookies = await this.cookies();
+    await this.doClearCookies();
+
+    const matches = (cookie: channels.NetworkCookie, prop: 'name' | 'domain' | 'path', value: string | RegExp | undefined) => {
+      if (!value)
+        return true;
+      if (value instanceof RegExp)
+        return value.test(cookie[prop]);
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/browserContext.ts:287]: reset lastIndex?

**Pull request merged**